docs/user/security/securing-communications/index.asciidoc

```diff
@@ -4,127 +4,184 @@
 <titleabbrev>Encrypting communications</titleabbrev>
 ++++
 
-{kib} supports Transport Layer Security (TLS/SSL) encryption for all forms of
-data-in-transit. Browsers send traffic to {kib} and {kib} sends traffic to {es}.
-These communications are configured separately.
+Secure Sockets Layer (SSL) and Transport Layer Security (TLS) provide encryption for data-in-transit. While these terms are often used
+interchangeably, {kib} only supports TLS, which has superseded the old SSL protocols.
+
+Browsers send traffic to {kib} and {kib} sends traffic to {es}. These communications are configured separately to use TLS.
+
+TLS requires X.509 certificates to perform encryption and authentication of the application that is being communicated with. {kib} supports
+certificates and private keys in PEM or PKCS#12 format.
 
 [[configuring-tls-browser-kib]]
 ==== Encrypting traffic between the browser and {kib}
 
-NOTE: You do not need to enable {security-features} for this type of encryption. 
-
-. Obtain a server certificate and private key for {kib}.
+. *Obtain a server certificate and private key for {kib}.*
 +
 --
-{kib} supports certificates/keys in both PKCS #12 key stores and PEM format.
-
-When you obtain a certificate, you must do at least one of the following:
+{kib} will need to use this "server certificate" and corresponding private key when receiving connections from web browsers.
 
-.. Set the certificate's `subjectAltName` to the hostname, fully-qualified domain name (FQDN), or IP address of the {kib} server.
+When you obtain a server certificate, you must set its subject alternative name (SAN) correctly to ensure that modern web browsers with
+hostname verification will trust it. You can set the SAN to the {kib} server's fully-qualified domain name (FQDN), hostname, or IP address.
 
-.. Set the certificate's Common Name (CN) to the {kib} server's hostname or FQDN. Using the server's IP address as the CN does not work.
-
-You may choose to generate a certificate and private key using {ref}/certutil.html[the {es} certutil tool]. If you already used certutil to
-generate a certificate authority (CA), you would generate a certificate/key for Kibana like so (using the `--dns` param to set the
-`subjectAltName`):
+You may choose to generate a certificate signing request (CSR) and private key using the {ref}/certutil.html[`elasticsearch-certutil`] tool.
+For example:
 
 [source,sh]
 --------------------------------------------------------------------------------
-bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12 --name kibana --dns localhost
+bin/elasticsearch-certutil csr -name kibana-server -dns <your_kibana_host>.com
 --------------------------------------------------------------------------------
 
-This will generate a certificate and private key in a PKCS #12 keystore named `kibana.p12`.
+This will produce a ZIP archive named `kibana-server.zip`. Extract that archive to obtain the PEM-formatted CSR (`kibana-server.csr`) and
```

Comment:
I think I made a comment previously regarding this, that we describe how to use elasticsearch-certutil to generate a self-signed ( or a signed by a custom auto generated CA ) certificate and use that in Kibana which will lead all browsers to throw a warning.

I think the idea with the CSR approach is that we show the users how to generate a CSR and then they take it to their organizational CA or to a public CA to get it signed so that the browsers trust the cert. From that perspective, it doesn't make sense to have instructions for signing the CSR, we're basically saying " hand this over to someone who knows what to do with it". 

If that is too complicated, we can fallback to the previous approach but make a note that users browsers won't like the resulting certificate and that if users want a trusted certificate, they need to talk to their org CA or a public trusted one. 

Or a combination of the two above approaches

Reply:
So, the old example was developer-focused -- it would "work", but you would have to install the CA certificate on your system to get your browser to trust the Kibana server. That simply isn't going to fit the use case for the vast majority of our users.

@jkakavas pointed out in another thread that this guide should really be focused on setting up Kibana for production use, ideally using a CSR. I tend to agree with him. I thought about going into more detail about what to do with the CSR once you've got it (you can either get a public CA to sign it, or use a private CA / internal PKI if your organization provides one). However, I decided that was creeping outside of the scope of this guide, and I didn't want to call out any for-profit public CAs in our documentation.

Of course, you don't *need* to get a CSR first -- you could go to any public CA and buy a certificate outright, but generating the CSR this way ensures that the attributes (CN, SAN) are set correctly and that the private key doesn't leave your system (unless you choose to move it).

Another idea would be to rewrite this (or add another section) to demonstrate how to obtain a publicly-signed certificate for Kibana using [Let's Encrypt](https://letsencrypt.org/), which is a free and open public CA. However, there are a lot of variables to the setup here depending on how your network is configured, and whether or not your Kibana server is exposed to the Internet. That's why I settled on "generate the CSR and leave it at that". Any experienced sysadmin should know what to do with a CSR.

I really struggled with this part, maybe @jkakavas has more feedback to share?

Reply:
Thanks @jkakavas and @jportner for explaining, the CSR makes more sense to me now. I agree that we should be offering best-practices here to secure a production cluster, so in light of that, I'm good with your changes.

Reply:
Oops, he beat me to the punch in responding!

> I think the idea with the CSR approach is that we show the users how to generate a CSR and then they take it to their organizational CA or to a public CA to get it signed so that the browsers trust the cert. From that perspective, it doesn't make sense to have instructions for signing the CSR, we're basically saying " hand this over to someone who knows what to do with it".

Exactly.

OK, sounds like we're in agreement and I'll leave it as-is.